### PR TITLE
sratoolkit 3.0.3

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  /tmp/sratoolkit-20230105-70930-o2gh7g/sra-tools-3.0.3/ncbi-vdb-source/interfaces/kfc/refcount.h:39:10: fatal error: 'atomic32.h' file not found
  #include <atomic32.h>
           ^~~~~~~~~~~~
  1 error generated.
```

**Comment by chenrui333** [comment]
managed to build with gcc-12, but not sure if that is the right route.

File: Formula/sratoolkit.rb [hunk 1]
```diff
@@ -38,13 +38,21 @@ class Sratoolkit < Formula
     end
   end
 
+  depends_on "bison" => :build
```
Comment (chenrui333) [inline Formula/sratoolkit.rb:41]: it needs newer bison, the system default is outdated.

File: Formula/sratoolkit.rb [hunk 2]
```diff
@@ -82,3 +95,28 @@ def install
...
+index d3edd6b..8323919 100644
+--- a/ngs/CMakeLists.txt
++++ b/ngs/CMakeLists.txt
+@@ -26,7 +26,4 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+ add_subdirectory( ncbi )
+ add_subdirectory( ngs-sdk )
+-if ( Java_FOUND )
+-	add_subdirectory( ngs-java )
+-endif()
```
Comment (chenrui333) [inline Formula/sratoolkit.rb:110]: I dont think we need java binding for now.

**Comment by chenrui333** [comment]
on linux

```
  Error: An exception occurred within a child process:
    Errno::ENOENT: No such file or directory @ dir_s_rmdir - /home/linuxbrew/.linuxbrew/Cellar/sratoolkit/3.0.3/bin/magic
```

**Comment by p-linnane** [comment]
For some reason the symlink that @cho-m put in before wasn't working any longer. I don't know if we need to use gcc or if we can sort that out instead.

File: Formula/sratoolkit.rb [hunk 3]
```diff
@@ -38,36 +38,36 @@ class Sratoolkit < Formula
...
   patch :DATA
 
   def install
+    ENV.cxx11
+
     (buildpath/"ncbi-vdb-source").install resource("ncbi-vdb")
 
     # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
     # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
     ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
```
Comment (cho-m) [inline Formula/sratoolkit.rb:60]: This is previous workaround I added to allow building ARM64 with clang. Upstream updated directory name to use `arm64` now - https://github.com/ncbi/ncbi-vdb/commit/01848812a06ac35528487221114d643402ec02b6

If you want to use `gcc` instead, then it should be restricted to ARM64 and should use compiler selection logic instead (`fails_with :clang`). Directly setting `CC`/`CXX` will skip the shims.

Comment (p-linnane) [reply Formula/sratoolkit.rb:60]: I don't know what we need to go down the `gcc` route if there is a simpler way to do this. Without the `gcc` commit we are still getting this:

```
fatal error: 'atomic32.h' file not found
#include <atomic32.h>
         ^~~~~~~~~~~~
1 error generated.
```

Comment (cho-m) [reply Formula/sratoolkit.rb:60]: Clang workaround would be:
```diff
@@ -41,8 +41,6 @@ class Sratoolkit < Formula
   depends_on "bison" => :build
   depends_on "cmake" => :build
   depends_on "doxygen" => :build
-
-  depends_on "gcc"
   depends_on "hdf5"
 
   uses_from_macos "flex" => :build
@@ -57,17 +55,12 @@ class Sratoolkit < Formula
 
     # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
     # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
-    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
+    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/arm64" if Hardware::CPU.arm?
 
     # Need to use HDF 1.10 API: error: too few arguments to function call, expected 5, have 4
     # herr_t h5e = H5Oget_info_by_name( self->hdf5_handle, buffer, &obj_info, H5P_DEFAULT );
     ENV.append_to_cflags "-DH5_USE_110_API"
 
-    gcc = Formula["gcc"]
-    gcc_major_ver = gcc.any_installed_version.major
-    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{gcc_major_ver}"
-    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{gcc_major_ver}"
-
     system "cmake", "-S", "ncbi-vdb-source", "-B", "ncbi-vdb-build", *std_cmake_args,
                     "-DNGS_INCDIR=#{buildpath}/ngs/ngs-sdk"
     system "cmake", "--build", "ncbi-vdb-build"
```

GCC usage with using compiler selection DSL would be:
```diff
@@ -41,13 +41,18 @@ class Sratoolkit < Formula
   depends_on "bison" => :build
   depends_on "cmake" => :build
   depends_on "doxygen" => :build
-
-  depends_on "gcc"
   depends_on "hdf5"
 
   uses_from_macos "flex" => :build
   uses_from_macos "libxml2"
 
+  on_macos do
+    on_arm do
+      depends_on "gcc"
+      fails_with :clang
+    end
+  end
+
   patch :DATA
 
   def install
@@ -55,19 +60,10 @@ class Sratoolkit < Formula
 
     (buildpath/"ncbi-vdb-source").install resource("ncbi-vdb")
 
-    # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
-    # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
-    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
-
     # Need to use HDF 1.10 API: error: too few arguments to function call, expected 5, have 4
     # herr_t h5e = H5Oget_info_by_name( self->hdf5_handle, buffer, &obj_info, H5P_DEFAULT );
     ENV.append_to_cflags "-DH5_USE_110_API"
 
-    gcc = Formula["gcc"]
-    gcc_major_ver = gcc.any_installed_version.major
-    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{gcc_major_ver}"
-    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{gcc_major_ver}"
-
     system "cmake", "-S", "ncbi-vdb-source", "-B", "ncbi-vdb-build", *std_cmake_args,
                     "-DNGS_INCDIR=#{buildpath}/ngs/ngs-sdk"
     system "cmake", "--build", "ncbi-vdb-build"
```

Comment (p-linnane) [reply Formula/sratoolkit.rb]: Looks like updating the workaround fixed us on macOS, but now Linux is failing.

**Comment by p-linnane** [comment]
Thank you @jonchang!

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**